packages/core-components/src/lib/unsorted/viz/table/_DataTable.svelte

```diff
@@ -297,19 +356,88 @@
 			  )
 			: data;
 
-	const weightedMean = (data, valueCol, weightCol) => {
-		let totalWeightedValue = 0;
-		let totalWeight = 0;
+	// ---------------------------------------------------------------------------------------
+	// GROUPED DATA
+	// ---------------------------------------------------------------------------------------
```

Comment:
This is a smell that we should try to break some items here out into functions in an external file